Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-problems.tsx

```diff
@@ -94,14 +102,37 @@ const ProblemsProvider = ({ children }: IProblemsProviderProps) => {
             const hashIndex = Number(hashParam) - 1;
             return normalizedProblems[hashIndex];
         },
-        [ normalizedProblems, hashParam ],
+        [ hashParam, normalizedProblems ],
     );
 
     const isLoadedFromHash = useMemo(
         () => !isNil(problemFromHash),
         [ problemFromHash ],
     );
 
+    const selectCurrentProblem = useCallback(
+        (problemId: number) => {
+            selectProblemById(problemId);
+
+            setInternalProblemId(null);
+        },
+        [ selectProblemById ],
+    );
+
+    const initiateInternalProblem = useCallback(
+        (problemId: number, contestId: number, participationType: ContestParticipationType) => {
+            const participateInContestUrl = getParticipateInContestUrl({
+                id: contestId,
+                participationType,
+            });
+
+            navigate({ pathname: participateInContestUrl });
+
+            setInternalProblemId(problemId);
```

Comment:
Can you add a comment describing the purpose of this function

Reply:
I am using it to redirect to contest page from another page (ex. From SearchProblem, or Contest Details Page) . I am setting the problemId which I will use to find the problem. I renamed it to initiateRedirectionToProblem for better understanding. Please let me know if you have a better naming suggestion.

Reply:
I meant comment in the code.